### DISCUSSION TITLE
Fearure Request:

### DISCUSSION THREAD

**Discussion by uowiec** [hassio-addons/app-zerotier#129] [Ideas]
# Problem/Motivation

Picture a home with LTE network connectivity, where the IP address granted from the mobile network is not routable. Zerotier is the only option. To make the home LAN available remotely, it is possible to configure a zerotier host as a NAT router, only there is no easy way to save iptables rules in the supervisor container.

## Expected behavior

I'd like to configure the RPi running HassOS and the zerotier plugin to be a gateway into home LAN, as described in this paper https://zerotier.atlassian.net/wiki/spaces/SD/pages/224395274/Route+between+ZeroTier+and+Physical+Networks

## Actual behavior

Following the instructions from the link above, I am able to create NAT in supervisor container, but I am not able to save the iptables rules to file, so after the HassOS restart the forwarding rules are lost.

## Steps to reproduce

Install the HassOS + ZeroTier plugin and follow the link above to configure NAT with iptables. You are now able to connect to your home LAN with ZeroTier.

## Proposed changes

I suggest adding an option to the configuration yaml like :
`nat: true`
alternatively, following these instructions https://zerotier.atlassian.net/wiki/spaces/SD/pages/193134593/Bridge+your+ZeroTier+and+local+network+with+a+RaspberryPi
`bridge: true`
but this might be an overkill. NAT might be enough as a starter.

**Comment by VitalyArtemiev** [comment]
Is it maybe possible to hack around this issue and run code that re-configures routing on startup?

**Comment by Sigmun** [comment]
I am in the exact same situation. I can connect to my HA server using zerotier add-on but not on my network through it. Too bad since zerotier can do it (https://zerotier.atlassian.net/wiki/spaces/SD/pages/193134593/Bridge+your+ZeroTier+and+local+network+with+a+RaspberryPi)!
